src/url.rs

```diff
@@ -0,0 +1,52 @@
+use urlparse::quote;
+
+pub fn get_url(src: &String, host: &str, compiler: &str, args: &String) -> String {
+    let codeeditor = json!(
+        {
+            "type": "component",
+            "componentName": "codeEditor",
+            "componentState": {
+                "id": 1,
+                "source": src,
+                "options": {
+                    "compileOnChange": true,
+                    "colouriseAsm": true
+                },
+                // "fontScale": 2.5
+            }
+        });
+
+    let compilerstate = json!(
+        {
+            "type": "component",
+            "componentName": "compiler",
+            "componentState": {
+                "source": 1,
+                "filters": {
+                    "commentOnly": true,
+                    "directives": true,
+                    "intel": true,
+                    "labels": true,
+                    "trim": true
+                },
+                "options": args,
+                "compiler": compiler,
+                // "fontScale": 3.0
```

Comment:
Speaking of easy copy/pasting, I think you left this in and didn't mean to :)

Reply:
Yes, that's true. There is so much that could be fine tuned :) I can rebase and remove this line...